core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1456,16 +1461,35 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
         DBIO.successful(RequestComplete(StatusCodes.OK, SubmissionValidationReport(submissionRequest, header, succeeded, failed)))
     }
 
-  def getSubmissionStatus(workspaceName: WorkspaceName, submissionId: String) = {
-    dataSource.inTransaction { dataAccess =>
+  def getSubmissionStatus(workspaceName: WorkspaceName, submissionId: String): Future[PerRequestMessage] = {
+    val submissionWithoutCosts = dataSource.inTransaction { dataAccess =>
       withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { workspaceContext =>
         withSubmission(workspaceContext, submissionId, dataAccess) { submission =>
           withUser(submission.submitter, dataAccess) { user =>
-            DBIO.successful(RequestComplete(StatusCodes.OK, new SubmissionStatusResponse(submission, user)))
+            DBIO.successful(submission, user)
           }
         }
       }
     }
+
+    submissionWithoutCosts flatMap {
+      case (submission, user) => {
+        val allWorkflowIds: Seq[String] = submission.workflows.flatMap(_.workflowId)
+        toFutureTry(submissionCostService.getWorkflowCosts(workspaceName.namespace, allWorkflowIds, GoogleProject(workspaceName.namespace))) map {
+          case Failure(_) =>
+            RequestComplete((StatusCodes.OK, new SubmissionStatusResponse(submission, user)))
```

Comment:
Should this write to logs in case of failure?

Reply:
yes, please log it.